app/src/main/java/de/westnordost/streetcomplete/location/LocationUtil.kt

```diff
@@ -0,0 +1,29 @@
+package de.westnordost.streetcomplete.location
+
+import android.Manifest
+import android.content.Context
+import android.content.IntentFilter
+import android.content.pm.PackageManager
+import android.location.Location
+import android.location.LocationManager
+
+import androidx.core.content.ContextCompat
+import androidx.core.location.LocationManagerCompat
+
+import de.westnordost.streetcomplete.data.osm.mapdata.LatLon
+
+fun isLocationEnabled(context: Context): Boolean {
+    return hasLocationPermission(context) && LocationManagerCompat
+            .isLocationEnabled(ContextCompat.getSystemService(context, LocationManager::class.java)!!)
+}
+
+fun hasLocationPermission(context: Context): Boolean {
+    return ContextCompat.checkSelfPermission(context,
+            Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+}
+
+fun createLocationAvailabilityIntentFilter(): IntentFilter {
+    return IntentFilter(LocationManager.MODE_CHANGED_ACTION)
+}
+
+fun Location.toLatLon(): LatLon = LatLon(latitude, longitude)
```

Comment:
If you like, you can also use the `fun xxx() = yyy` pattern (instead of the classic function declaration), it is widely used in the app.

Reply:
Ha, I meant to do that (a reformatting commit), but that was where I stopped last night and I forgot about it when I picked it up this morning :upside_down_face: